mono/mini/exceptions-amd64.c

```diff
@@ -900,15 +900,15 @@ mono_arch_handle_altstack_exception (void *sigctx, MONO_SIG_HANDLER_INFO_TYPE *s
 	 *   return ip
 	 * 128 is the size of the red zone
 	 */
-	frame_size = sizeof (MonoContext) + sizeof (gpointer) * 4 + 128;
+	frame_size = sizeof (ucontext_t) + sizeof (gpointer) * 4 + 128;
 	frame_size += 15;
 	frame_size &= ~15;
 	sp = (gpointer *)(UCONTEXT_REG_RSP (sigctx) & ~15);
 	sp = (gpointer *)((char*)sp - frame_size);
-	copied_ctx = (MonoContext*)(sp + 4);
+	copied_ctx = (ucontext_t*)(sp + 4);
 	/* the arguments must be aligned */
 	sp [-1] = (gpointer)UCONTEXT_REG_RIP (sigctx);
-	mono_sigctx_to_monoctx (sigctx, copied_ctx);
+	memcpy (sigctx, copied_ctx, sizeof (ucontext_t));
```

Comment:
it's swapped, it should be `memcpy (copied_ctx, sigctx, sizeof (...))`

Reply:
Of course, messed it up just before committing...